### ISSUE TITLE
Doesn't seem to be able to both send and receive locally via local address

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1811]
**_Reported by Paul Sokolovsky:_**

I'm doing following:

1. Create a UDP socket and bind it to a known local address (192.0.2.1), attach recv callback.
2. Create another UDP socket, and send data to first socket's address (192.0.2.1:PORT).
3. All this happens on the same device (specifically qemu_x86, specifically with Zephyr.js).

What I get is: socket <span>#</span>1 receives nothing, and socket <span>#</span>2 doesn't even have sent callback called. Whereas expected behavior is: socket <span>#</span>1 receive the data.

Is it supposed to work in Zephyr now? If not, that's not what POSIX (or common sense) wants. I'm looking into this because it's an easy way to perform automated integration testing for networking stuff (we can use it in sanitycheck tests).
 

(Imported from Jira ZEP-1966)

**Comment by nashif** [comment]
**_by Jukka Rissanen:_**

Indeed this should work properly. We have check to localhost address (127.0.0.1) which should work ok and you should get the packet back, but we seem to be missing the check to our own address. This looks like a simple one liner to fix, I will investigate more.

**Comment by nashif** [comment]
**_by Paul Sokolovsky:_**

Verified with my Zephyr.js, works perfect. Actually, I verified 2 cases: sending and receiving via a single context, and via 2 separate contexts. Thanks!

For reference, the JS code was:


{code:java}
var PORT = 33333;
var HOST = '192.0.2.1';

var dgram = require('dgram');

var server = dgram.createSocket('udp4');
server.on('message', function (message, remote) {
    console.log(remote.address + ':' + remote.port +' - ' + message.toString("ascii"));
});
server.bind(PORT, HOST);

var client = dgram.createSocket('udp4');
var buf = new Buffer("foo");
client.send(buf, 0, buf.length, PORT, HOST);

console.log("End");
```
